### ISSUE TITLE
Bug with formgrader and sqlalchemy

### ISSUE THREAD

**Issue by amillb** [jupyter/nbgrader#1434]
### Operating system
Mac OS 10.14.6

### `nbgrader --version`
0.6.1

### `jupyterhub --version` (if used with JupyterHub)
N/A

### `jupyter notebook --version` 
6.2.0

### Expected behavior
Clicking on Manage Assignment -> the number of submissions leads to the list of students and options to auto grade, etc.

### Actual behavior
The notebook hangs with the "loading please wait" message and the following terminal output. 

The problem is resolved by downgrading sqlalchemy from 1.4.7 to 1.3.23. (I found this out because `nbgrader` was working correctly in my other Python environment.

```
[E 16:11:55.815 NotebookApp] Uncaught exception GET /formgrader/api/submissions/wk1_census (::1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/formgrader/api/submissions/wk1_census', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/tornado/web.py", line 1702, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/tornado/web.py", line 3173, in wrapper
        return method(self, *args, **kwargs)
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/nbgrader/server_extensions/formgrader/base.py", line 108, in wrapper
        return f(self, *args, **kwargs)
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/nbgrader/server_extensions/formgrader/base.py", line 117, in wrapper
        return f(self, *args, **kwargs)
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/nbgrader/server_extensions/formgrader/apihandlers.py", line 167, in get
        submissions = self.api.get_submissions(assignment_id)
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/nbgrader/apps/api.py", line 587, in get_submissions
        db_submissions = gb.submission_dicts(assignment_id)
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/nbgrader/api.py", line 3053, in submission_dicts
        all_scores = aliased(union_all(
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/sqlalchemy/orm/util.py", line 1201, in aliased
        return AliasedClass(
      File "/Users/adammb/anaconda3/envs/urbandatascience/lib/python3.8/site-packages/sqlalchemy/orm/util.py", line 495, in __init__
        mapper = insp.mapper
    AttributeError: 'CompoundSelect' object has no attribute 'mapper'
[E 16:11:55.819 NotebookApp] {
      "Host": "localhost:8888",
      "Accept-Language": "en-us",
      "Accept-Encoding": "gzip, deflate",
      "Cookie": "username-localhost-8888=\"2|1:0|10:1618441910|23:username-localhost-8888|44:NDQyMDUyNTE5OTY4NGVjNWI4ZmQ2N2EwZGU4NTkxNjg=|675d1a7c803b89ce73633387c18c602f41553ab5e30ee7d7535aca402465b0f2\"; username-localhost-8890=\"2|1:0|10:1618359623|23:username-localhost-8890|44:ZTJlOWRlNDU1Y2Y5NDdlNjk1MmU2N2RiOTVlMTM4OTc=|cf147e27f5172a643583097f7287676f7cef24f438702033816061b36a5b9720\"; username-localhost-8889=\"2|1:0|10:1618332202|23:username-localhost-8889|44:OGQxZDRkOTUwMWM1NDIyMjgwM2ViNmEzNTk5NTFlNjQ=|7ef18c96b0b10c3a9ebdc21dcf3c73821ee7c515daad9ee4333f4cbc16d213ae\"; username-localhost-8891=\"2|1:0|10:1618282756|23:username-localhost-8891|44:NzMxYjg1MDRmZTE5NDRhMDg4N2RhN2I0NjhjNDU0OTE=|3df5fe0a4d23fa55e2603ec11256a926517afcded022cad2737fb7fc251b60df\"; username-localhost-8787=\"2|1:0|10:1617733726|23:username-localhost-8787|44:OTJjZWFhM2JiYjg2NGQ3ZGJjZmQ0Yzc3ZDc5MzM4Yzk=|ca53cd86d1edd4c396adb4b51c6169265b2be7f114ffdbfab6fa82bb65115098\"; username-localhost-8893=\"2|1:0|10:1617731551|23:username-localhost-8893|44:YmI2OTYwZDM4YzdlNDIwZmJmNWNkYTZjNDNlMDNlM2I=|7108634d24d60c796200f4ad6476d59fd741254de70ffd21686470f11d109772\"; username-localhost-8900=\"2|1:0|10:1617725750|23:username-localhost-8900|44:MmNlMGU0YmY1N2RlNDY5MThiNWQwYzE5MDgzNzEyYmI=|efad2a3cfdc8b8ba40cb22884bb4c9d97547b76deb51f4611ee39bfec87557b3\"; username-localhost-8808=\"2|1:0|10:1617681047|23:username-localhost-8808|44:ZmY4NTQ2YmIyZmFkNGQyZWJhMjQ5MzI4ZWUwYTQ4OGI=|b44118306801a4976dede9ed53767d3f5cf3ae61c045c5bdef6f3588f913b183\"; username-localhost-8947=\"2|1:0|10:1617663440|23:username-localhost-8947|44:ZjEzYjgwNzAyZjkwNDE1MDlkYWExYWQ1ZmZhNWQxMjM=|815f712e82a98e6596096a64ab8f257b8421afd73580c2b206129606a7b8fe35\"; username-localhost-8898=\"2|1:0|10:1617635388|23:username-localhost-8898|44:ZmI5ODQ0YTUyM2RmNGRjY2JhYzczZTU5ZjBmZjNmZWY=|b858116cd65afd043489348748a7a5bfafda322978832e971cc6a4688300eecd\"; username-localhost-8967=\"2|1:0|10:1617396922|23:username-localhost-8967|44:YzU3YzYxOTljZjY4NDQxMWFiOTIwYjFiYmJjZmUwZDg=|0dc0f9985b00f7b29206a04e91386db00680cd4f9d355964b01cfdfadbde5888\"; username-localhost-8990=\"2|1:0|10:1617392747|23:username-localhost-8990|44:ZmVlYjYxOTVjMjU1NDY1NGFlMTZmMjVmODYzZTFkNGQ=|dc7d6106c571f61e85c4f0f870ea49182caebb33278304e81a43a3ed901aea57\"; username-localhost-8912=\"2|1:0|10:1617392245|23:username-localhost-8912|44:ODBkMTgyOGEwODdjNGY4MmFjMjIwMjFkMjRhN2NlMDQ=|9f474d4cd364f6ff4cdc21e8e578e9eae07d2c90a8413111c69b0ba8218c19f7\"; username-localhost-8892=\"2|1:0|10:1617386557|23:username-localhost-8892|44:NGIyYzZmODc4YzE3NDMzMWEyNTJhMDI2MzhjYmVjNjc=|2586605069bbb180ae371bc7190ae3060cab02f2638cd7055a8079709c747d08\"; _xsrf=2|fee58af6|32c6d6849138c360d0a4a33296717ffe|1617292833; username-localhost-8903=\"2|1:0|10:1617152533|23:username-localhost-8903|44:NjEzMTAyYTQzNGQwNDIxMmExMDdiNmJlMzI3YTZhMGM=|4008e09e8d44162862b861308550448a34b6fa3ac57273aec6d2cf70f12d73ce\"",
      "Connection": "keep-alive",
      "Accept": "application/json, text/javascript, */*; q=0.01",
      "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15",
      "X-Csrftoken": "2|fee58af6|32c6d6849138c360d0a4a33296717ffe|1617292833",
      "Referer": "http://localhost:8888/formgrader/manage_submissions/wk1_census",
      "X-Requested-With": "XMLHttpRequest"
    }
[E 16:11:55.819 NotebookApp] 500 GET /formgrader/api/submissions/wk1_census (::1) 11.650000ms referer=http://localhost:8888/formgrader/manage_submissions/wk1_census
```

### Steps to reproduce the behavior
Clicking on Manage Assignment

**Comment by carlosnatalino** [comment]
I was also having the same problem. I had installed nbgrader from pip.

I downloaded it and installed it directly from the repository and it is now working fine: https://nbgrader.readthedocs.io/en/stable/contributor_guide/installation_developer.html

**Comment by vahtras** [comment]
I have had the same problem. Thanks for the tip to downgrade sqlalchemy!

**Comment by jhamrick** [comment]
This is fixed on master in #1420, so I will close this issue.
